edge/pkg/metamanager/metamanager_test.go

```diff
@@ -59,19 +59,13 @@ func TestNameAndGroup(t *testing.T) {
 }
 
 func TestStart(t *testing.T) {
-	modules := core.GetModules()
-	for name, module := range modules {
-		if name == MetaManagerModuleName {
-			metaModule = module
-			break
-		}
-	}
-	core.Register(&metaManager{enable: true})
-	if metaModule == nil {
-		t.Errorf("failed to register to beehive")
-	}
+	beehiveContext.InitContext(beehiveContext.MsgCtxTypeChannel)
+	mt := newMetaManager(true)
+	core.Register(mt)
+	beehiveContext.AddModule(mt.Name())
+	beehiveContext.AddModuleGroup(mt.Name(), mt.Group())
 
-	go metaModule.Start()
+	go mt.Start()
```

Comment:
Looks good, actually if we move `core.Register(&metaManager{enable: true})` above to the for loop, the panic issue may can also be avoided. @muxuelan  And the InitContext had been called in init func